test/page.spec.js

```diff
@@ -48,19 +48,29 @@ module.exports.addTests = function({testRunner, expect, headless, Errors, Device
       await newPage.close();
       expect(await browser.pages()).not.toContain(newPage);
     });
-    it_fails_ffox('should run beforeunload if asked for', async({context, server}) => {
+    it('should run beforeunload if asked for', async({context, server}) => {
       const newPage = await context.newPage();
       await newPage.goto(server.PREFIX + '/beforeunload.html');
       // We have to interact with a page so that 'beforeunload' handlers
       // fire.
       await newPage.click('body');
-      newPage.close({ runBeforeUnload: true });
+      const pageClosingPromise = newPage.close({ runBeforeUnload: true });
       const dialog = await waitEvent(newPage, 'dialog');
       expect(dialog.type()).toBe('beforeunload');
-      expect(dialog.defaultValue()).toBe('');
-      expect(dialog.message()).toBe('');
-      dialog.accept();
-      await waitEvent(newPage, 'close');
+      if (CHROME) {
+        expect(dialog.defaultValue()).toBe('');
```

Comment:
Are these not implemented in Firefox?

Reply:
the `message()` is intentionally empty in Chrome for beforeunload - to avoid scam.

Updated the with the actual ffox text.